Validate the size limit in BufferedTokenizer.

mashhurs opened the pull request:
## Release notes
[rn:skip]

## What does this PR do?
Currently, `BufferedTokenizer` doesn't validate size limit. That means any negative or zero value can lead `input buffer full` exception. This PR adds a safeguard validation on size limit to allow only positive number. 

## Why is it important/What is the impact to the user?
As it is a public interface, this enhancement improves interface experience.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## How to test this PR locally


## Related issues

- Closes #16824

## Use cases

## Screenshots

## Logs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/2030)
* Commit: cc36be318039a0a4b4fe11ed7fe30e18e0f7e8d8

<!--buildkite-pr-comment-logstash-pull-request-pipeline
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/2030","number":2030,"commit":"cc36be318039a0a4b4fe11ed7fe30e18e0f7e8d8"}],"number":2030}
buildkite-pr-comment-->

mashhurs commented on logstash-core/src/main/java/org/logstash/common/BufferedTokenizerExt.java:
```diff
@@ -55,7 +55,11 @@ public IRubyObject init(final ThreadContext context, IRubyObject[] args) {
         if (args.length == 2) {
-            this.sizeLimit = args[1].convertToInteger().getIntValue();
+            final int sizeLimit = args[1].convertToInteger().getIntValue();
+            if (sizeLimit <= 0) {
```
We had talks/ideas to ignore setting the limit if specific value (for example negative -1) or zero. It has an interface accepts a single delimiter to ignore limit setting. So, I would like _only_ to make this limit validated and not add any other hidden features.

yaauie replied:
I agree.

As it stands, this PR only prevents BufTok from being initialized with values that could _never_ result in bytes being emitted :+1:

yaauie reviewed (approved):
LGTM 👍🏼 

Backport targets: `8.x`, `8.17`, `8.16`

mashhurs commented:
@logstashmachine backport 8.x

mashhurs commented:
@logstashmachine backport 8.17

mashhurs commented:
@logstashmachine backport 8.16

mashhurs commented:
@logstashmachine backport 8.16

mashhurs commented:
Interestingly there is a conflict with 8.16 (but it looks same with 8.17?!), will manually backport:
```
Run python devtools/backport 8.16 16882 --remote=origin --yes
  
HEAD is now at a21510103 Validate the size limit in BufferedTokenizer. (#16882)
Already on 'main'
Your branch is up to date with 'origin/main'.
Already up to date.
Switched to a new branch '[8](https://github.com/elastic/logstash/actions/runs/12700618738/job/35403640695#step:12:9).16'
branch '8.16' set up to track 'origin/8.16'.
Already up to date.
error: branch 'backport_16882_8.16' not found
Switched to a new branch 'backport_16882_8.16'
CONFLICT (modify/delete): logstash-core/benchmarks/src/main/java/org/logstash/benchmark/BufferedTokenizerExtBenchmark.java deleted in HEAD and modified in a215[10](https://github.com/elastic/logstash/actions/runs/12700618738/job/35403640695#step:12:11)103 (Validate the size limit in BufferedTokenizer. (#16882)).  Version a21510103 (Validate the size limit in BufferedTokenizer. (#16882)) of logstash-core/benchmarks/src/main/java/org/logstash/benchmark/BufferedTokenizerExtBenchmark.java left in tree.
error: could not apply a2[15](https://github.com/elastic/logstash/actions/runs/12700618738/job/35403640695#step:12:16)10103... Validate the size limit in BufferedTokenizer. (#[16](https://github.com/elastic/logstash/actions/runs/12700618738/job/35403640695#step:12:17)882)
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Namespace(commit_hashes=[], continue=False, diff=False, from_branch='main', pr_number='16882', remote='origin', to_branch='8.16', yes=True)
```
